impl/c-qube/src/services/csv-adapter/parser/utils/csvreader.ts

```diff
@@ -3,13 +3,20 @@ const fs = require('fs').promises;
 
 import * as csv from 'csv-parser';
 
-export async function readCSV(filePath: string): Promise<string[][]> {
+export async function readCSV(filePath: string, configPath?: string, delimiter?: string): Promise<string[][]> {
```

Comment:
don't pass in `configPath` to a utility function like this. Read the required paramters (`delimiter` in this case) at the initial top level when the config is parsed and pass in the delimiter from there only.